### PR TITLE
[Subs 1181] fill uploaded files field in submission envelope

### PR DESCRIPTION
**Pull request by ke4**

### CODE REVIEW

File: src/main/java/uk/ac/ebi/subs/processing/dispatcher/DispatcherServiceImpl.java [hunk 1]
```diff
@@ -177,6 +188,33 @@ public void insertReferencedSamples(SubmissionEnvelope submissionEnvelope) {
...
     }
 
+    @Override
+    public void insertUploadedFiles(SubmissionEnvelope submissionEnvelope) {
+        String submissionId = submissionEnvelope.getSubmission().getId();
+
+        List<UploadedFile> uploadedFiles = submissionEnvelope.getAssayData().stream().map(
+                AssayData::getFiles
+        ).collect(Collectors.toList())
+            .stream().flatMap(List::stream).collect(Collectors.toList())
```
Comment (davidrichardson) [inline src/main/java/uk/ac/ebi/subs/processing/dispatcher/DispatcherServiceImpl.java]: do we need to collect to a list here - can we carry on working on the stream?

Comment (ke4) [reply src/main/java/uk/ac/ebi/subs/processing/dispatcher/DispatcherServiceImpl.java]: Agreed

File: src/main/java/uk/ac/ebi/subs/processing/dispatcher/DispatcherServiceImpl.java [hunk 2]
```diff
@@ -177,6 +188,33 @@ public void insertReferencedSamples(SubmissionEnvelope submissionEnvelope) {
...
+                .stream().map(
+                    fileRef -> {
+                        String fileName = fileRef.getName();
+                        File file = fileRepository.findByFilenameAndSubmissionId(fileName, submissionId);
```
Comment (davidrichardson) [inline src/main/java/uk/ac/ebi/subs/processing/dispatcher/DispatcherServiceImpl.java]: This has the potential to be a lot of database lookups - could we get all the uploaded files in one go?

Comment (ke4) [reply src/main/java/uk/ac/ebi/subs/processing/dispatcher/DispatcherServiceImpl.java]: Thanks for this catch and perfectly agree. Bad code 👎

Comment (ke4) [reply src/main/java/uk/ac/ebi/subs/processing/dispatcher/DispatcherServiceImpl.java]: Rewrote

File: src/main/java/uk/ac/ebi/subs/processing/dispatcher/DispatcherServiceImpl.java (hunk 1 above)
Comment (ke4) [reply src/main/java/uk/ac/ebi/subs/processing/dispatcher/DispatcherServiceImpl.java]: Removed

**Comment by davidrichardson** [comment]
👍

**Pull request merged**